src/System.Memory/tests/Span/Sort.cs

```diff
@@ -0,0 +1,1606 @@
+        }
+
+        [Theory]
+        [Trait(SortTrait, SortTraitValue)]
+        [MemberData(nameof(s_fastSortTests))]
+        public static void Sort_Keys_Int32(ISortCases sortCases)
+        {
+            Test_Keys_Int32(sortCases);
+        }
+
+        [Theory]
+        [Trait(SortTrait, SortTraitValue)]
+        [MemberData(nameof(s_fastSortTests))]
+        public static void Sort_Keys_UInt32(ISortCases sortCases)
+        {
+            Test_Keys_UInt32(sortCases);
+        }
+
+        [Theory]
+        [Trait(SortTrait, SortTraitValue)]
+        [MemberData(nameof(s_fastSortTests))]
+        public static void Sort_Keys_Int64(ISortCases sortCases)
+        {
+            Test_Keys_Int64(sortCases);
+        }
+
+        [Theory]
+        [Trait(SortTrait, SortTraitValue)]
+        [MemberData(nameof(s_fastSortTests))]
+        public static void Sort_Keys_UInt64(ISortCases sortCases)
+        {
+            Test_Keys_UInt64(sortCases);
+        }
+
+        [Theory]
+        [Trait(SortTrait, SortTraitValue)]
+        [MemberData(nameof(s_fastSortTests))]
+        public static void Sort_Keys_Single(ISortCases sortCases)
+        {
+            Test_Keys_Single(sortCases);
+        }
+
+        [Theory]
+        [Trait(SortTrait, SortTraitValue)]
+        [MemberData(nameof(s_fastSortTests))]
+        public static void Sort_Keys_Double(ISortCases sortCases)
+        {
+            Test_Keys_Double(sortCases);
+        }
+
+        [Theory]
+        [Trait(SortTrait, SortTraitValue)]
+        [MemberData(nameof(s_fastSortTests))]
+        public static void Sort_Keys_Boolean(ISortCases sortCases)
+        {
+            Test_Keys_Boolean(sortCases);
+        }
+
+        [Theory]
+        [Trait(SortTrait, SortTraitValue)]
+        [MemberData(nameof(s_fastSortTests))]
+        public static void Sort_Keys_Char(ISortCases sortCases)
+        {
+            Test_Keys_Char(sortCases);
+        }
+
+        [Theory]
+        [Trait(SortTrait, SortTraitValue)]
+        [MemberData(nameof(s_fastSortTests))]
+        public static void Sort_Keys_String(ISortCases sortCases)
+        {
+            Test_Keys_String(sortCases);
+        }
+
+        [Theory]
+        [Trait(SortTrait, SortTraitValue)]
+        [MemberData(nameof(s_fastSortTests))]
+        public static void Sort_Keys_ComparableStructInt32(ISortCases sortCases)
+        {
+            Test_Keys_ComparableStructInt32(sortCases);
+        }
+
+        [Theory]
+        [Trait(SortTrait, SortTraitValue)]
+        [MemberData(nameof(s_fastSortTests))]
+        public static void Sort_Keys_ComparableClassInt32(ISortCases sortCases)
+        {
+            Test_Keys_ComparableClassInt32(sortCases);
+        }
+
+        [Theory]
+        [Trait(SortTrait, SortTraitValue)]
+        [MemberData(nameof(s_fastSortTests))]
+        public static void Sort_Keys_BogusComparable(ISortCases sortCases)
+        {
+            Test_Keys_BogusComparable(sortCases);
+        }
+#if OUTER_LOOP
```

Comment:
Why do we need this #if?

Reply:
@ahsonkhan we don't this is just for me sharing code and doing tests outside of corefx, a lot of the nits are due to reasons of getting a faster feedback loop and being able to swap different implementations etc. I am cleaning this up now.